File: frontend/src/types/candidate.ts
export interface Candidate {
  id: number;
  name: string;
  email: string | null;
  phone: string | null;
  resume_score: number; // JD match score %
  overall_resume_score?: number; // Weighted resume score
  grade?: string;
  experience_years: number;
  education_score: number;
  certifications: number;
  prediction: string; // 'Shortlist' | 'Needs Review' | 'Reject'
  confidence?: number;
  rank_score?: number;
  matched_skills: string[];
  missing_skills: string[];
  created_at: string;
  rank?: number;
  filtered_rank?: number;
}

export interface Evaluation {
  id: number;
  candidate_id: number;
  question: string;
  answer: string;
  score: number;
  technical_depth: number;
  communication: number;
  completeness: number;
  strengths: string[];
  improvements: string[];
  created_at: string;
}

export interface KPIStats {
  total: number;
  shortlisted: number;
  needs_review: number;
  rejected: number;
  average_score: number;
}

export interface SkillStats {
  skill: string;
  count: number;
}

export interface ScoreBucket {
  range: string;
  count: number;
}

export interface ExperienceScorePlot {
  name: string;
  experience: number;
  score: number;
  prediction: string;
}

export interface AnalyticsResponse {
  kpis: KPIStats;
  prediction_distribution: { name: string; value: number }[];
  top_skills: SkillStats[];
  top_missing_skills: SkillStats[];
  score_distribution: ScoreBucket[];
  experience_vs_score: ExperienceScorePlot[];
}

File: backend/app/schemas/candidate.py
"""
Pydantic schemas validation for Candidate screening models.
"""

from datetime import datetime
from typing import List, Optional
from pydantic import BaseModel


class CandidateBase(BaseModel):
    name: str
    email: Optional[str] = None
    phone: Optional[str] = None
    resume_score: float
    matched_skills: List[str]
    missing_skills: List[str]
    experience_years: float
    education_score: float
    certifications: int
    prediction: str
    projects_count: int = 2


class CandidateCreate(CandidateBase):
    pass


class CandidateOut(CandidateBase):
    id: int
    created_at: datetime

    class Config:
        from_attributes = True
